File: java/src/main/java/com/meteroid/models/CouponFilter.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum CouponFilter implements ToQueryParam {
    ALL("ALL"),
    ACTIVE("ACTIVE"),
    INACTIVE("INACTIVE"),
    ARCHIVED("ARCHIVED");
    private final String value;

    CouponFilter(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: go/coupon_filter.go
// this file is @generated
package meteroid

type CouponFilter string

// Known values of CouponFilter.
//
// The API may add new values over time; unknown values round-trip unchanged.
const (
	CouponFilterAll      CouponFilter = "ALL"
	CouponFilterActive   CouponFilter = "ACTIVE"
	CouponFilterInactive CouponFilter = "INACTIVE"
	CouponFilterArchived CouponFilter = "ARCHIVED"
)

// AllCouponFilterValues lists every CouponFilter value known to this SDK version.
var AllCouponFilterValues = []CouponFilter{
	CouponFilterAll,
	CouponFilterActive,
	CouponFilterInactive,
	CouponFilterArchived,
}

// String returns the wire representation of the value.
func (e CouponFilter) String() string {
	return string(e)
}

// IsKnown reports whether the value is one this SDK version knows about.
func (e CouponFilter) IsKnown() bool {
	for _, known := range AllCouponFilterValues {
		if e == known {
			return true
		}
	}
	return false
}
